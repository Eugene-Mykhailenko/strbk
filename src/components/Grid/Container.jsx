import React from 'react';
import PropTypes from 'prop-types';
import styles from './Grid.scss';

const Container = (props) => {
  const { children } = props;

  return <div className={styles.container}>{children}</div>;
};

Container.propTypes = {
  /** Special property to pass children
   * elements directly into component output. */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Container;
