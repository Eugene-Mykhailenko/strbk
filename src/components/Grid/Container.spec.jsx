import React from 'react';
import Container from './Container';

const div = <div className="find-me">Lorem Ipsum is simply dummy text</div>;

describe('Container', () => {
  it('should render with no children by default', () => {
    const container = shallow(<Container>{div}</Container>);

    expect(container.prop('className')).toEqual('container');
    expect(container).toMatchSnapshot();
  });

  it('should render provided children', () => {
    const container = shallow(<Container>{div}</Container>);

    expect(container.find('.find-me')).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
