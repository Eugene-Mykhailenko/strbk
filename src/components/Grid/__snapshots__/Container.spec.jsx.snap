// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render provided children 1`] = `
<div
  className="container"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;

exports[`Container should render with no children by default 1`] = `
<div
  className="container"
>
  <div
    className="find-me"
  >
    Lorem Ipsum is simply dummy text
  </div>
</div>
`;
